connector/openshift/openshift.go

```diff
@@ -231,6 +284,8 @@ func newHTTPClient(insecureCA bool, rootCA string) (*http.Client, error) {
 		if !tlsConfig.RootCAs.AppendCertsFromPEM(rootCABytes) {
 			return nil, fmt.Errorf("no certs found in root CA file %q", rootCA)
 		}
+	} else if certPool != nil {
+		tlsConfig = tls.Config{RootCAs: certPool}
```

Comment:
Note, it is neither documented nor obvious that a set `RootCA` causes `TrustedCertPool` to be ignored. Preferably, these should be combined (might be hard to accomplish because a `x509.CertPool` can neither be cloned nor the certificates extracted from it, and we should avoid modifying the object supplied by the user); alternatively, an error should be raised (as early as possible, i.e., in `Open`) when both are set; and the third (and least preferred) alternative is to ignore `RootCA` if `TrustedCertPool` is set (`TrustedCertPool` is the lower-level type - it's easy to convert a `RootCA` string into a `TrustedCertPool`, but hard or even impossible to do the reverse; for this reason, `TrustedCertPool` should get preference).

However, also see the comment above regarding a boolean flag for system roots.